src/conn.rs

```diff
@@ -478,6 +507,42 @@ impl<'a, 'c> InProgress<'a, 'c> {
     }
 }
 
+struct InProgressTransactWatcher<'a> {
+    cache_watcher: InProgressCacheTransactWatcher<'a>,
+    observer_watcher: InProgressObserverTransactWatcher<'a>,
+    tx_id: Option<Entid>,
+}
+
+impl<'a> InProgressTransactWatcher<'a> {
+    fn new(observer_watcher: InProgressObserverTransactWatcher<'a>, cache_watcher: InProgressCacheTransactWatcher<'a>) -> Self {
+        InProgressTransactWatcher {
+            cache_watcher: cache_watcher,
+            observer_watcher: observer_watcher,
+            tx_id: None,
+        }
+    }
+}
+
+impl<'a> TransactWatcher for InProgressTransactWatcher<'a> {
+    type Result = ();
```

Comment:
All of these seem to have `Result = ()` — do we still need this?

Reply:
nah, that was left over from a previous experiment. I'll revert.